eclair-core/src/main/scala/fr/acinq/eclair/db/WalletDb.scala

```diff
@@ -0,0 +1,29 @@
+package fr.acinq.eclair.db
+
+import fr.acinq.bitcoin.{BinaryData, BlockHeader, Transaction}
+import fr.acinq.eclair.blockchain.electrum.ElectrumClient.GetMerkleResponse
+
+trait HeaderDb {
+  def addHeader(height: Int, header: BlockHeader): Unit
+
+  def addHeaders(startHeight: Int, headers: Seq[BlockHeader]): Unit
+
+  def getHeader(height: Int): Option[BlockHeader]
+
+  // TODO: should not be needed
+  def getHeader(blockHash: BinaryData): Option[(Int, BlockHeader)]
+
+  def getHeaders(startHeight: Int, maxCount: Option[Int]): Seq[BlockHeader]
+
+  def getTip: Option[(Int, BlockHeader)]
+}
+
+trait TransactionDb {
+  def addTransaction(tx: Transaction, proof: GetMerkleResponse): Unit
+
+  def getTransaction(txid: BinaryData): Option[(Transaction, GetMerkleResponse)]
+
+  def getTransactions(): Seq[(Transaction, GetMerkleResponse)]
+}
+
+trait WalletDb extends HeaderDb with TransactionDb
```

Comment:
`WalletDb` is too generic given that it is in the package `fr.acinq.eclair.db`

Reply:
I'll move WalletDb to the electrum package